metrics/cli/test/flutter/adapter/flutter_cli_service_adapter_test.dart

```diff
@@ -55,14 +56,48 @@ void main() {
     test(
       ".build() enables web support before building the web application",
       () async {
-        await flutterCliServiceAdapter.build(path);
+        await flutterAdapter.build(path);
 
         verifyInOrder([
           flutterCli.enableWeb(),
           flutterCli.buildWeb(path),
         ]);
       },
     );
+
+    test(
+      ".version() rethrows the StateError if showing the version throws it",
+          () {
+        when(flutterCli.version()).thenThrow(stateError);
+
+        final actual = expectAsync0(() => flutterAdapter.version());
+
+        expect(actual, throwsStateError);
+      },
+    );
+
+    test(
+      ".build() doesn't build the web application if enabling web support throws the StateError ",
```

Comment:
It seems like it doesn't matter which error was thrown from the `CLI`, the `build` method does not build the application in any case.
```suggestion
      ".build() doesn't build the web application if enabling web support throws",
```